source/js/buyers-guide/bg-main.js

```diff
@@ -194,6 +212,34 @@ let main = {
         injectDonateModal(donationModal, modalOptions);
       }
     */
+
+    if (document.querySelectorAll(`.join-us`)) {
+      var elements = Array.from(document.querySelectorAll(`.join-us`));
+
+      if (elements.length) {
+        elements.forEach(element => {
+          var props = element.dataset;
+
+          props.apiUrl = `${networkSiteURL}/api/campaign/signups/${props.signupId ||
+            0}/`;
+
+          props.csrfToken = props.csrfToken || csrfToken;
+
+          apps.push(
+            new Promise(resolve => {
+              ReactDOM.render(
+                <JoinUs
+                  {...props}
+                  isHidden={false}
```

Comment:
should we move this into props, too, since we're already working with a separate var?
```
var props = element.dataset;
...
props.isHidden = false;
props.whenLoaded = () => resolve();

apps.push( ... <JoinUs {...props}/>, element);
```

Reply:
Not sure why but if I do `props.whenLoaded = resolve;` it actually transforms `resolve` function into a string. 👀 

```js
console.log(typeof props.whenLoaded); // returns string
```